### PR TITLE
Add Basic Documentation for Terms Lookup Query

### LINKED ISSUE #430: [FEATURE] documentation for terms lookup queries
**Is your feature request related to a problem?**
opensearch supports terms lookup queries, but they're currently undocumented.

**What solution would you like?**
extend the [terms query documentation](https://opensearch.org/docs/latest/opensearch/query-dsl/term/#terms) to include terms lookup queries.

**What alternatives have you considered?**
n/a

**Do you have any additional context?**
see the original [documentation from ES](https://www.elastic.co/guide/en/elasticsearch/reference/7.10/query-dsl-terms-query.html#query-dsl-terms-lookup).

### PR DESCRIPTION
**Pull request by rursprung**
### Description
this adds very rudimentary documentation for TLQ so that it at least
doesn't go unmentioned. however, there are many more details which
should be mentioned. this should then be done with #430.

more information can be found in the [Elasticsearch TLQ
Documentation][ESTLQ]. i'm not 100% sure whether this was under the
Apache 2.0 license in 7.10 (the repo states that everything is Apache
2.0 unless explicitly stated otherwise in the file and the documentation
files don't seem to include any other license?) and thus didn't simply
take over the documentation from there (which would of course be the
easiest and most complete way of doing it).

also i'm not sure (and didn't find this information anywhere) whether
the test data actually exists and the queries can be run against the
test data? in that case the provided example either needs to be adapted
to the existing test data or the test data needs to be extended
accordingly (i doubt that an index `play-assignments` already exists?).
due to this the example has also not been tested.

Signed-off-by: Ralph Ursprung <Ralph.Ursprung@avaloq.com>

[ESTLQ]: https://www.elastic.co/guide/en/elasticsearch/reference/7.10/query-dsl-terms-query.html#query-dsl-terms-lookup
 
### Issues Resolved
partially resolves #430 (but more will be required to fully resolve it, but i'll leave that to someone else with better knowledge about TLQ as the ES docs show that it has some details to be aware of)
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by rursprung** [comment]
> Hey @rursprung. We'll have to modify this more to make it OpenSearch specific. As a general rule of thumb, we can't reuse any documentation from a third-party, even under an Apache 2.0 license.

just for clarification: i wrote 100% of the content of this PR w/o taking it over from ES (i actually refrained from re-reading their docs before writing this PR to avoid accidentally taking anything over and only controlled it afterwards) which is also why it's incomplete.

> Would you be okay if I took over this PR from here?

absolutely, go ahead! thanks a lot!

**Comment by alicejw-aws** [comment]
Hi @rursprung ,
I also have some changes to the same section 
( Query DSL) that will compliment your changes for this PR. I've made some minor suggestions in your PR that align with the other changes. 
Please take a look at this PR: https://github.com/opensearch-project/documentation-website/pull/561
I hope it makes sense to view both PR changes together. Thanks

File: _opensearch/query-dsl/term.md [hunk 1]
```diff
@@ -271,6 +271,33 @@ GET shakespeare/_search
...
 You get back documents that match any of the terms.
 
+### Terms Lookup
+
+You can use a `terms` query with a `lookup` to match values based on a field in a document in another index.
+
+Parameter | Behavior
+:--- | :---
+`index` | The index from which the document is read.
+`id` | The id of the documented.
```
Comment (keithhc2) [inline _opensearch/query-dsl/term.md:281]: Typo here. I'm guessing it should be "document" instead of "documented"?

**Review by keithhc2 — APPROVED** [review]

**Review by lizsnyder — APPROVED** [review]

File: _opensearch/query-dsl/term.md [hunk 2]
```diff
@@ -271,6 +271,33 @@ GET shakespeare/_search
...
+`path` | Path to the field from which the values are used in the terms query.
+
+E.g. to get all lines from the shakespeare play for a role (or roles) specified in the index `play-assignments` for the entry `42`:
```
Comment (lizsnyder) [inline _opensearch/query-dsl/term.md:284]: Tiny nit, maybe capitalize Shakespeare

**Review by alicejw-aws — COMMENTED** [review]
Hi @rursprung ,
I also have some changes to the same section
( Query DSL) that will compliment your changes for this PR. I've made some minor suggestions in your PR that align with the other changes.
Please take a look at this PR: https://github.com/opensearch-project/documentation-website/pull/561
I hope it makes sense to view both PR changes together. Thanks

**Comment by rursprung** [comment]
@alicejw-aws: since you're editing the exact same files i guess this would lead to merge conflicts for the 2nd PR after merging the first (regardless of which one goes in first). it might make sense for you to just take over the changes i proposed here directly into your PR to avoid having to rebase / merge afterwards?

File: _opensearch/query-dsl/term.md [hunk 3]
```diff
@@ -271,6 +271,33 @@ GET shakespeare/_search
 
 You get back documents that match any of the terms.
 
+### Terms Lookup
```
[the remark this answers is not in the document]
Comment (rursprung) [reply _opensearch/query-dsl/term.md:274]: IMHO this is wrong. there's a difference between `term` queries and `terms` queries and only `terms` supports `lookup`, so i would strongly advise against this change in nomenclature!

**Comment by alicejw-aws** [comment]
Hi @rursprung, thanks for looking out for merge conflicts, i do appreciate that. However, the changes I suggested here for you are separate (yet complimentary and need to be entered either in this PR or in a separate one). 

If you prefer to have a separate PR for my changes to this file, I can do that. (i was trying to save time and effort by doing it with this PR).

**Comment by alicejw-aws** [comment]
Ok. The plan is to merge this PR without my proposed changes, then add my changes to the separate [561 PR](https://github.com/opensearch-project/documentation-website/pull/561)

no worries. this should work out with the same outcome: consistent complimentary information that contains the uniform terms throughout the doc set (e.g. Term Lookup Query TQL).

**Comment by rursprung** [comment]
ah, i think we misunderstood each other: my proposal was to just abandon this PR and take over everything you want from this directly in #561 - not to merge this one here w/o your suggestion (otherwise i'd have applied them)

**Comment by alicejw-aws** [comment]
oh ok thanks. no worries. i acted on your previous response, so it's merged now. my PR is in-review for the changes i had suggested previously. hopefully i can get those merged today and it will be ok.

**Pull request merged**